### ISSUE TITLE
Listeners OnMeAction only processed AFTER text is sent

### ISSUE THREAD

**Issue by DarthGandalf** [kvirc/KVIrc#154]
**Reported by bpressler on 30 May 2008 18:37:55 UTC**
I try to place an event listener on OnMeAction. The listener is deigned to replace certain text in a CTCP ACTION and only send out the changed text to the server.

However what happenes is, that first the unchanged text is sent t the server, and then the changed text.
On client side I only SEE the changed line. It looks like jsut the display of the unchanged line is surpressed, but it is still sent.

The code for the Listener is like this:

**\* code start
%replace = $hash( \
    "m##","(Marica)",\
    "na#","(Nadia)",\
    "ay#","(Aya)",\
    "ki#","(Kiani)",\
    "no#","(Nomad)",\
    "sy#","(Sylea)",\
    "ta#","(Tanya)",\
    "ya#","(Illyana)",\
    "ke#","(Keowyn)",\
    "mi#","(Mina)",\
)
**\* code end

foreach (%key,$keys(%replace)){
if ($str.contains($0,%key)){
                me $str.replace($0,%replace{%key},%key);
                halt;
}}

Expected:

I type: /me #mm test

Text sent: MyNick (Marica) test

Text seen on client side: MyNick (Marica) test

Actual result:

I type: /me #mm test

Text sent: MyNick m## test
Text sent: MyNick (Marica) test

Text seen on Client side: MyNick (Marica) test

Additional Information:

The same kind of code for the evenListener works like a charm, when used on the event OnTextInput and inside the code "say" is used instead of "me".

I run KVirc 3.4.0 under windows XP.

---

Migrated from: https://svn.kvirc.de/kvirc/ticket/154

**Comment by DarthGandalf** [comment]
**Comment by bpressler on 30 May 2008 18:39:14**
Umm - for better viewing the actual code, you can look here:
http://pastebin.ca/1033452

**Comment by DarthGandalf** [comment]
**Comment by almostautomated on 05 Jul 2008 04:20:04**
confirmed...

here's a workaround that works for channel windows (not query); it substitutes mme for /me

```
if ($str.left($0,3) == "mme") {
    %szNewMsg=$str.stripleft($str.rightfromfirst($0," "))

    %replace = $hash("a##","(AlmostAutomated)","aa#","(AlmostAutomated_)")
    foreach (%key,$keys(%replace)) {
        if ($str.contains(%szNewMsg,%key)) {
            %szNewMsg = $str.replace(%szNewMsg,%replace{%key},%key);
        }
    } 
    ctcp $chan.name action %szNewMsg
    echo -i=$msgtype(action) -w=$active %szNewMsg
    halt
}
```

Also; using onTextInput doesn't seem to trigger when using /say from the command input line as the documentation states; that is unless you use the -x flag.  Which seems opposite from the doc.

**Comment by DarthGandalf** [comment]
**Comment by ctrlaltca on 08 Aug 2008 12:02:58**
Fixed in 2215 (branch 3.4), 2216 (trunk 4.0)

---

Resolution: none → fixed
Status: new → closed
